docs: add module documentation for Registry Cache (#120)
kyma-project/registry-cache: docs/user/troubleshooting/01-10-incorrect-credentials.md

# Image Pulls Fail with "404 manifest unknown" Despite Correct Image Name

## Symptom

> Image pulls from a cached upstream registry fail consistently with `404 manifest unknown` errors, even though the image exists in the upstream registry and the image name is correct.

Reviewer: The problem with current implementation is that the user will not directly see when their configuration is incorrect. They may think everything is OK, as the Registry Cache is using fallback (`imagePullSecrets`) if credentials referenced in `RegistryCacheConfig` are incorrect. I'm wondering how we should approach that. When using protected registries you actually should check logs of the pod created in `kube-system` by the Gardener. It sound a little bit awkward - dear user please see manually if it works :) . At some point it could be automated, but with the current design to be 100% sure the protected registry caching works we would need to check the logs.

## Cause

The upstream registry returns `404` instead of `401` when credentials are incorrect. This makes a credential failure indistinguishable from a missing image at the log level.

## Solution

1. Check the Registry Cache Pod logs for the affected upstream:

   ```bash

>    kubectl logs -n kube-system -l app=registry-cache --tail=50

Reviewer: incorrect `app=registry-cache` label

> # Image Pulls Fail with "404 manifest unknown" Despite Correct Image Name

Reviewer: This troubleshooting is meant for the following case:
- User configured RegistryCache with credentials
- They observe/or suspect that the image caching functionality doesn't work

## Symptom

Image pulls from a cached upstream registry fail consistently with `404 manifest unknown` errors, even though the image exists in the upstream registry and the image name is correct.

> ### Note:

> > Registry Cache is designed to not impair operations if its configuration is incorrect. If you have configured an `imagePullSecret` on your workloads (recommended), image pulls will still succeed via direct fallback to the upstream registry even when Registry Cache credentials are wrong. This means image pull failures may not be visible even with misconfigured credentials — the only way to verify the cache is working correctly is to check the registry cache Pod logs as described below.

Reviewer 2: ```suggestion
> Registry Cache is designed to not impair operations if its configuration is incorrect. If you have configured an `imagePullSecret` on your workloads (recommended), image pulls still succeed using direct fallback to the upstream registry even when the Registry Cache credentials are incorrect. This means image pull failures may not be visible even with misconfigured credentials — the only way to verify the cache is working correctly is to check the registry cache Pod logs as described below.
```

> > Registry Cache is designed to not impair operations if its configuration is incorrect. If you have configured an `imagePullSecret` on your workloads (recommended), image pulls will still succeed via direct fallback to the upstream registry even when Registry Cache credentials are wrong. This means image pull failures may not be visible even with misconfigured credentials — the only way to verify the cache is working correctly is to check the Gardener extension Pod logs as described below.

Reviewer: Gardener extension Pod logs is a bit confusing here. In the next section we "use registry cache Pod" which is more precise and its meaning is clear from the context.

## Cause

> The upstream registry returns `404` instead of `401` when credentials are incorrect. This makes a credential failure indistinguishable from a missing image at the log level.

Reviewer: This can be unclear to the user. This sentence is IMO lacking important context.
"The upstream registry returns `404`" refers to registry cache pod in the `kube-system` namespace.

> When Registry Cache credentials are incorrect, the registry cache Pod in `kube-system` receives a `404` response from the upstream registry instead of `401`. This makes a credential failure indistinguishable from a missing image at the log level. Meanwhile, image pulls in workloads continue to succeed via the `imagePullSecret` fallback, so the misconfiguration is not immediately visible.

Reviewer: I would modify this section a bit. 
From the Gardener docs:
_The registry cache provides the credentials for every request against the corresponding upstream. In some cases, misconfigured credentials can prevent the registry cache to pull even public images from the upstream (for example: invalid service account key for Artifact Registry). However, this behaviour is controlled by the server-side logic of the upstream registry._

Maybe we should change this piece: "_receives a `404` response from the upstream registry instead of `401`. This makes a credential failure indistinguishable from a missing image at the log level." The fact we expect 401 and get 404 doesn't add much (what is important: when we see 404 it should )_"

When testing with Artifactory I received `404`. But it may be different for other upstream registries. To be on the safe side I would state that we are expecting to see some error an refer to `404` as an example

> When Registry Cache credentials are incorrect, the registry cache Pod in `kube-system` receives an error response from the upstream registry. Depending on the upstream registry, the error may be indistinguishable from a missing image at the log level — for example, Artifact Registry returns `404` instead of an authentication error. Meanwhile, image pulls in workloads continue to succeed using the `imagePullSecret` fallback, so the misconfiguration is not immediately visible.

Reviewer: Actually I verified that JFrog Artifactory (I referred to that as Artifactory) returns 404. I'm not sure about Artifact Registry.
